biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -1118,6 +1120,21 @@ public Manifest calcManifest() throws Exception {
 				requirements.add(ExecutionEnvironmentNamespace.EXECUTION_ENVIRONMENT_NAMESPACE, attrs);
 			}
 
+			if (isTrue(getProperty(NOSERVICELOADERREGISTRAR))) {
```

Comment:
This would be the wrong place. You want to stop it from being added rather that delete it after the fact. Also, this would remove any user defined caps from the bnd file.